backend/dal/Helpers/Extensions/BuildingExtensions.cs

```diff
@@ -23,5 +27,105 @@ public static void ThrowIfNotUnique(this DbSet<Entity.Building> buildings, Entit
+            filter.ThrowIfNull(nameof(user));
+
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = user.GetAgencies();
+            var viewSensitive = user.HasPermission(Permissions.SensitiveView);
+
+            // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+            var query = context.Buildings.AsNoTracking().Where(b =>
+                !b.IsSensitive || (viewSensitive && userAgencies.Contains(b.AgencyId)));
+
+            if (filter.NELatitude.HasValue && filter.NELongitude.HasValue && filter.SWLatitude.HasValue && filter.SWLongitude.HasValue)
+                query = query.Where(b =>
+                    b.Latitude != 0 &&
+                    b.Longitude != 0 &&
+                    b.Latitude <= filter.NELatitude &&
+                    b.Latitude >= filter.SWLatitude &&
+                    b.Longitude <= filter.NELongitude &&
+                    b.Longitude >= filter.SWLongitude);
+
+            if (filter.Agencies?.Any() == true)
+            {
+                // Get list of sub-agencies for any agency selected in the filter.
+                var agencies = filter.Agencies.Concat(context.Agencies.AsNoTracking().Where(a => filter.Agencies.Contains(a.Id)).SelectMany(a => a.Children.Select(ac => ac.Id)).ToArray()).Distinct();
+                query = query.Where(p => agencies.Contains(p.AgencyId));
+            }
+            if (filter.ClassificationId.HasValue)
+                query = query.Where(p => p.ClassificationId == filter.ClassificationId);
+            if (filter.StatusId.HasValue)
+                query = query.Where(p => p.StatusId == filter.StatusId);
+            if (!String.IsNullOrWhiteSpace(filter.ProjectNumber))
+                query = query.Where(p => EF.Functions.Like(p.ProjectNumber, $"{filter.ProjectNumber}%"));
+            if (!String.IsNullOrWhiteSpace(filter.Description))
+                query = query.Where(p => EF.Functions.Like(p.Description, $"%{filter.Description}%"));
+            if (filter.ConstructionTypeId.HasValue)
+                query = query.Where(b => b.BuildingConstructionTypeId == filter.ConstructionTypeId);
+            if (filter.PredominateUseId.HasValue)
+                query = query.Where(b => b.BuildingPredominateUseId == filter.PredominateUseId);
+            if (filter.FloorCount.HasValue)
+                query = query.Where(b => b.BuildingFloorCount == filter.FloorCount);
+            if (!String.IsNullOrWhiteSpace(filter.Tenancy))
+                query = query.Where(b => EF.Functions.Like(b.BuildingTenancy, $"%{filter.Tenancy}%"));
+
+            if (!String.IsNullOrWhiteSpace(filter.Address)) // TODO: Parse the address information by City, Postal, etc.
+                query = query.Where(b => EF.Functions.Like(b.Address.Address1, $"%{filter.Address}%") || EF.Functions.Like(b.Address.City.Name, $"%{filter.Address}%"));
```

Comment:
Should this also be parsing the city name using the address query field?

Reply:
I believe so, otherwise "Victoria" won't return much.